mp/src/game/client/game_controls/SpectatorGUI.cpp

```diff
@@ -783,6 +792,47 @@ void CSpectatorGUI::UpdateTimer()
 	SetLabelText("timerlabel", szText );
 }
 
+#ifdef NEO
+//-----------------------------------------------------------------------------
+// Purpose: Updates the spectated player's health
+//-----------------------------------------------------------------------------
+void CSpectatorGUI::UpdatePlayerLabel()
+{
+	IGameResources* gr = GameResources();
+	int specmode = GetSpectatorMode();
+	int playernum = GetSpectatorTarget();
+
+	// update player name filed, text & color
+	if (playernum > 0 && playernum <= gpGlobals->maxClients && gr)
+	{
+		Color c = gr->GetTeamColor(gr->GetTeam(playernum)); // Player's team color
+
+		m_pPlayerLabel->SetFgColor(c);
+
+		wchar_t playerText[80], playerName[64], health[10];
```

Comment:
`playerName[MAX_PLAYER_NAME_LENGTH+1]` should suffice - since `MAX_PLAYER_NAME_LENGTH` equals 32, allocating 64*sizeof(wchar_t) bytes is probably more than we need here?

Could also zero-initialize statically as `playerName[MAX_PLAYER_NAME_LENGTH+1]{0}` instead of using the memset, although perhaps the compiler will optimise it regardless.

Reply:
![image](https://github.com/user-attachments/assets/490c28b7-8a86-4d4f-9c82-7bd7c9a778a3)

But I guess it would make more sense to use that?

Reply:
Well, looking at the release disassembly it gets optimized into a single stack allocation, so at most there's gonna be a 32 byte padding there which I suppose isn't a big deal.